app/javascript/app/form-validation.js

```diff
@@ -10,26 +10,33 @@ document.addEventListener('DOMContentLoaded', () => {
   }
 
   if (forms.length !== 0) {
-    [].forEach.call(forms, function(form) {
+    [].forEach.call(forms, function (form) {
       const buttons = form.querySelectorAll('[type="submit"]');
-      form.addEventListener('submit', function() {
-        if (buttons.length !== 0) {
-          [].forEach.call(buttons, function (button) {
-            button.disabled = true;
-          });
-        }
-        const submitSpinner = document.getElementById('submit-spinner');
-        if (submitSpinner) {
-          submitSpinner.className = '';
-        }
-      }, false);
+      form.addEventListener(
+        'submit',
+        function () {
+          if (buttons.length !== 0) {
+            [].forEach.call(buttons, function (button) {
+              button.disabled = true;
+            });
+          }
+          const submitSpinner = document.getElementById('submit-spinner');
+          if (submitSpinner) {
+            submitSpinner.className = '';
+          }
+        },
+        false,
+      );
       const elements = form.querySelectorAll('input');
       if (elements.length !== 0) {
-        [].forEach.call(elements, function(input) {
+        [].forEach.call(elements, function (input) {
           input.addEventListener('input', function () {
             if (buttons.length !== 0 && input.checkValidity()) {
-              [].forEach.call(buttons, function(button) {
-                if (button.disabled && !button.classList.contains('no-auto-enable')) {
+              [].forEach.call(buttons, function (button) {
+                if (
+                  button.disabled &&
+                  !button.classList.contains('no-auto-enable')
+                ) {
```

Comment:
This is a very weird style to me, I'm more used to something like....

```js
if (button.disabled &&
    !button.classList.contains('no-auto-enable')) {
  button.disabled = false;
}
```

Or extracting a variable to keep line length short:

```js
var shouldEnable = button.disabled &&
  !button.classList.contains('no-auto-enable');
if (shouldEnable) {
  button.disabled = false;
}
```

What's the name of the rule that does this? Are we open to changing it?

Reply:
Another thought.... a lot of these changes seem to be because of line width.

WDYT of increasing max line width to 100chars? That's what our [current rubocop config](https://github.com/18F/identity-idp/blob/master/.rubocop.yml#L52-L56) is and it seems a good candidate for parity across languages?

Reply:
> What's the name of the rule that does this? Are we open to changing it?

This is one of those points where as an opinionated code formatter, we'd have to choose to abandon personal preferences to whatever the tool decides. There are [some Prettier options](https://prettier.io/docs/en/options.html), though very few, and this is not one of them. So in effect it becomes a choice of accepting the formatting style or not using Prettier.

To the specific style in question: I've personally encountered both varieties (and in previous work was [involved in championing a change from the former to the latter](https://make.wordpress.org/core/2018/12/18/proposed-revisions-to-javascript-coding-standards/)). I've become more fond of the Prettier variation, since it adds some predictability in one-condition-per-line, and in indirectly encouraging variable extraction (as you mention) in cases where one would want to try to avoid the multi-line conditional.

In this case, the variable extraction example you mention would be formatted in a very similar way though:

```js
var shouldEnable = (
  button.disabled &&
  !button.classList.contains('no-auto-enable')
);

if (shouldEnable) {
  button.disabled = false;
}
```

You could do some gymnastics to try to avoid it:

```js
const { disabled, classList } = button;
const isEnabled = disabled && !classList.contains('no-auto-enable');
if ( isEnabled ) {
  button.disabled = false;
}
```

Reply:
> Another thought.... a lot of these changes seem to be because of line width.
> 
> WDYT of increasing max line width to 100chars? That's what our [current rubocop config](https://github.com/18F/identity-idp/blob/master/.rubocop.yml#L52-L56) is and it seems a good candidate for parity across languages?

[We could change it](https://prettier.io/docs/en/options.html#print-width). I'm not of a strong-held opinion here. I tend to prefer shorter line lengths as _intentional friction_ to force a developer to write code more expressively within those constraints.

But consistency with the Rubocop configuration is a compelling argument.

Reply:
Ok, if this is "the prettier way" I guess we may as well keep it.... but how about that 100c line length? 😀

Reply:
The note at the above link is pretty relevant here, though. It describes how Prettier will _strive to meet_ the print width, rather than treat it as an absolute maximum.

Reply:
As discussed directly, let's plan to go with 100:

- Alignment with Rubocop
- It's also [the Airbnb standard recommendation](https://github.com/airbnb/javascript#whitespace--max-len)
- It's reasonable to assume that most standard editor widths should have support to show more than 80 columns
- We might encounter issues with using complex types in JSDoc where 80 column width would be prohibitive

I'll push this update shortly.

Reply:
Updated in the rebased ae47c08.